client/signup/steps/domains/index.jsx

```diff
@@ -123,8 +123,28 @@ class DomainsStep extends Component {
 		}
 		this.setCurrentFlowStep = this.setCurrentFlowStep.bind( this );
 		this.state = {
+			...this.state,
 			currentStep: null,
+			shouldSkipStep: 'loading',
 		};
+		if ( this.state.shouldSkipStep === true ) {
+			this.props.goToNextStep();
+		}
+		this.domainExperiment = loadExperimentAssignment( 'onboarding_pm_skip_domain' );
+	}
+
+	componentWillMount() {
```

Comment:
This lifecycle hook should not be used.

https://legacy.reactjs.org/docs/react-component.html#unsafe_componentwillmount